torch_xla/amp/syncfree/grad_scaler.py

```diff
@@ -0,0 +1,11 @@
+import torch
+
+
+class GradScaler(torch.cuda.amp.GradScaler):
```

Comment:
Question, does `GradScaler` really needs to be in a separate class under `syncfree`? Would be make sense to add a `sync_free` option to existing amp `GradScalar`?

Reply:
Using one `GradScaler` is a good idea. There are two options: 
1. Add a `sync_free` option to existing `amp.GradScaler.__init__`.
2. Add a check like`is_syncfree_optim = "found_inf" in inspect.signature(optimizer.step).parameters`.

I think probably option 2 is better as it automatically detects if the optimizer is sync_free optimizer.